### DISCUSSION TITLE
How to update golang in multiple files (go.mod, Dockerfile, etc)?

### DISCUSSION THREAD

**Discussion by nitrocode** [renovatebot/renovate#28402] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Here's an example pr of upgrading golang from v1.22.1 to v1.22.2.

https://github.com/runatlantis/atlantis/pull/4405

It only updates go.mod and errors out because there is a mismatch between the golang version in go.mod vs the Dockerfile

https://github.com/runatlantis/atlantis/blob/a357b377ff42869b7756efaef57c179a5d9762a2/Dockerfile#L5

https://github.com/runatlantis/atlantis/blob/a357b377ff42869b7756efaef57c179a5d9762a2/go.mod#L3

https://github.com/runatlantis/atlantis/blob/a357b377ff42869b7756efaef57c179a5d9762a2/testing/Dockerfile#L1

**Comment by rarkins** [comment]
An update for that Dockerfile is pending. If you look in your Dependency Dashboard you can see `chore(deps): update golang docker tag to v1.22.2 in testing/dockerfile (main)`

These are not automatically grouped together because one has depName=go and the other has depName=golang, so they get different branch names. If you add a package rule with matchPackageNames=["go", "golang"] plus groupName=go then you should see them in the same PR

**Reply by nitrocode** [reply]
Hi rarkins. Thanks for the quick response. A package rule like this?

```
    {
      matchDatasources: [
        'go',
        'golang-version'
      ],
      matchPackageNames: [
        'go',
        'golang',
      ],
      versioning: 'go',
      groupName: 'go'
    },
```

Ref https://github.com/runatlantis/atlantis/pull/4435

**Reply by rarkins** [reply]
Yes although you probably don't need matchDatasources. If it's not working then create a minimal reproduction we can more easily debug

**Reply by nitrocode** [reply]
I tried the above and it seems to half work where it updated the image but it updated the image from 1.22.1-alpine to 1.22.2 without the alpine suffix which caused it to fail the build since without the suffix, it defaults to Debian

PR: https://github.com/runatlantis/atlantis/pull/4456/files

Config: https://github.com/runatlantis/atlantis/blob/main/.github/renovate.json5

Is this a bug?

**Reply by rarkins** [reply]
No, it's wrong config. Don't set go versioning for docker images

**Reply by nitrocode** [reply]
I figured I have something misconfigured which is why I was requesting help in this discussion. If you have a sec, please point me in the right direction as I do not understand where to go from here. 

Are you suggesting that I exclude the dockerfile update and instead manually update it instead of allowing renovate to update it? Or is there a way for me to configure it so it doesn't remove the alpine suffix? Perhaps if I do something like this?

```dockerfile
ARG GOLANG_TAG=1.22.1

FROM golang:${GOLANG_TAG}-alpine
```

I very much appreciate your time.

**Reply by rarkins** [reply]
My assumption was that this line is wrong and should be removed: https://github.com/runatlantis/atlantis/blob/d825d5894e41ede81e77cd6331108e2aeae6c7f0/.github/renovate.json5#L79

By default Renovate will retain suffixes (e.g. -alpine) so long as "docker" versioning is used.
